io/net/virt-net/Network_virtualization.py

```diff
@@ -0,0 +1,311 @@
+        '''
+        Removing Backing device for Network virtualized device
+        '''
+        pre_remove = self.backing_dev_count(self.slot_num)
+        self.backing_dev_add_remove(self.server, 'r', self.lpar_id,
+                                    self.slot_num, self.add_backing_device,
+                                    'remove')
+        post_remove = self.backing_dev_count(self.slot_num)
+        if pre_remove - post_remove != 1:
+            self.fail("Failed to remove backing device")
+
+    def test_network_virtualization_remove(self):
+        '''
+        Network virtualized device remove operation
+        '''
+        self.network_virtualization_add_remove_device(self.server, 'r',
+                                                      self.lpar_id,
+                                                      self.slot_num,
+                                                      self.backing_devices,
+                                                      'remove')
+        output = self.network_virtualization_list_device(self.server,
+                                                         self.lpar,
+                                                         self.slot_num)
+        if 'slot_num=%s' % self.slot_num in str(output):
+            self.log.debug(output)
+            self.fail("lshwres still lists the Network virtualized device \
+                       after remove operation")
+
+    def network_virtualization_add_remove_device(self, server, operation,
+                                                 lpar_id, slot_num,
+                                                 backing_devices, msg):
+        '''
+        Adds and removes a Network virtualized device based
+        on the operation
+        '''
+        if operation == 'a':
+            cmd = 'chhwres -m %s --id %s -r virtualio --rsubtype vnic \
+                   -o a -s %s -a \"%s\" '\
+                   % (server, lpar_id, slot_num, backing_devices)
+        else:
+            cmd = 'chhwres -m %s --id %s -r virtualio --rsubtype vnic \
+                   -o r -s %s'\
+                   % (server, lpar_id, slot_num)
+        try:
+            cmd = self.run_command(cmd)
+        except CommandFailed as cf:
+            self.log.debug(str(cf))
+            self.fail("Network virtualization %s device operation \
+                       failed" % msg)
+
+    def network_virtualization_list_device(self, server, lpar, slot_num):
+        '''
+        Lists the Network vritualized devices
+        '''
+        cmd = 'lshwres -r virtualio -m %s --rsubtype vnic --filter \
+              \"lpar_names=%s,slots=%s\"' % (server, lpar, slot_num)
+        try:
+            output = self.run_command(cmd)
+            print output
+        except CommandFailed as cf:
+            self.log.debug(str(cf))
+            self.fail("lshwres operation failed ")
+        return output
+
+    def backing_dev_add_remove(self, server, operation, lpar_id, slot_num,
```

Comment:
Most of the parameters are already class parameters here, can be read via `self`. So no need to pass them.